apps/contracts/contracts/Bandada.sol

```diff
@@ -10,18 +10,41 @@ contract Bandada is IBandada, Ownable {
     /// @dev See {IBandada-groups}.
     mapping(uint256 => uint256) public override groups;
 
+    mapping(uint256 => mapping(uint256 => uint256))
+        public fingerprintCreationDates;
+
+    mapping(uint256 => uint256) public override fingerprintDuration;
+
     /// @dev See {IBandada-updateGroups}.
     function updateGroups(
         Group[] calldata _groups
     ) external override onlyOwner {
         for (uint256 i = 0; i < _groups.length; ) {
             groups[_groups[i].id] = _groups[i].fingerprint;
 
+            fingerprintCreationDates[_groups[i].id][
+                _groups[i].fingerprint
+            ] = block.timestamp;
+
             emit GroupUpdated(_groups[i].id, _groups[i].fingerprint);
 
             unchecked {
                 ++i;
             }
         }
     }
+
+    function getFingerprintCreationDate(
```

Comment:
Doesn't this function already exist as a mapping function? I think it doesn't need the `onlyOwner` modifier anyway